public/js/navigator-js/integration/StateViewMap.js

```diff
@@ -1,111 +1,161 @@
+        j, state, statesLength,
+        viewInstance, $reactRoot;
+
+      for (index = 0; index < recipesLength; index++) {
+        recipe = this._orderedRecipes[index];
+        recipeStates = recipe.getStates();
+        statesLength = recipeStates.length;
+
+        for (j = 0; j < statesLength; j++) {
+          state = recipeStates[j];
+
+          if (requestedState.contains(state)) {
+
+            this._addViewElementToDOM(recipe);
+            var viewInstance = recipe.getViewInstance();
+            if (Array.isArray(viewInstance.navigatorBehaviors)) {
+              this._navigator.add(viewInstance, state);
+            }
+          }
+        }
+      }
+    },
+
+    _addRecipeToParent: function _addRecipeToParent(parentRecipe, recipe) {
+      var $container = this._$root,
+        $inside,
+        insideSelector = recipe.getInsideSelector();
+      var _recipe = recipe;
+      if (parentRecipe) {
+        $container = parentRecipe.getRootEl();
+      }
+
+      if (insideSelector != null) {
+        $inside = $container.find(insideSelector);
+        $container = $inside.length > 0 ? $inside.first() : $container;
+      }
+
+      var additionType = '' + (parentRecipe ? parentRecipe._type : 'BACKBONE') + ' > ' + recipe._type;
+
+      switch (additionType) {
+        case 'BACKBONE > BACKBONE':
+          var i = this._orderedRecipes.indexOf(recipe) + 1,
+          length = this._orderedRecipes.length,
+          testRecipe;
+
+          for (i; i < length; i++) {
+            testRecipe = this._orderedRecipes[i];
+
+            // If any other views have the same parent, add this element before
+            // those elements in the container element
+
+            if (testRecipe.isInstantiated() && testRecipe.isMounted() && testRecipe.getRootEl().parent()[0] === $container[0]) {
+              testRecipe.getRootEl().before(recipe.getRootEl());
+              return;
+            }
+          }
+          $container.append(recipe.getRootEl());
+          break;
+
+        case 'BACKBONE > REACT':
+          $reactRoot = $(document.createElement('div'));
+          $reactRoot.addClass('react-root');
+          $container.append($reactRoot);
+          ReactDOM.render(recipe._viewInstance, $reactRoot[0]);
+
+          break;
+
+        case 'REACT > REACT':
+          parentRecipe._showChild(recipe);
+
+          // Find root react element
+          while (parentRecipe.getParentRecipe() && parentRecipe.getParentRecipe()._type === 'REACT') {
+            parentRecipe = parentRecipe.getParentRecipe();
+          }
+          // TODO: Batch this render call on state change
+          ReactDOM.render(
+            parentRecipe._viewInstance,
+            parentRecipe.getRootEl().parent()[0]
+          );
+          break;
+        default:
+          console.error('Invalid recipe type combination: ' + additionType);
+          break;
+
+      }
+    },
+
+    _addViewElementToDOM: function _addViewElementToDOM(recipe) {
```

Comment:
This function was changed in that the actual addition of a recipe to the DOM was split into a seperate function, and this function just verifies that the recipe is initialized and that it needs mounting.